### PR TITLE
add unit test skeleton

### PR DESCRIPTION
**Pull request by ciuffagianluca**
#### List of Changes
Added unit test and pipeline step to execute them

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/__tests__/HealthControllers.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ERROR_QUEUE_NAME: "error q name",
+    INITIAL_RETRY_TIMEOUT_SECONDS: 10,
+    MAX_RETRY_ATTEMPTS: 3,
+    PORT: 3240,
+    RETRY_QUEUE_NAME: "retry q name",
+    STORAGE_CONNECTION_STRING: "storageconnection"
+  } as configuration.IConfig;
+
+
+    it("should return a error response",async () => {
```
Comment (giovanniberti) [inline src/controllers/__tests__/HealthControllers.test.ts:46]:
```suggestion
    it("should return a error response", async () => {
```

File: src/controllers/__tests__/HealthControllers.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      var handler = HealthControllers.getHealth(config, logger);
+
+      var req2 = {} as any;
+
+      const responseErrorValidation = await handler(req2);
```
Comment (giovanniberti) [inline src/controllers/__tests__/HealthControllers.test.ts:51]:
```suggestion
      var request = {} as any;

      const responseErrorValidation = await handler(request);
```

File: src/controllers/__tests__/InfoControllers.test.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    // tslint:disable-next-line: no-console
+    warn: jest.fn().mockImplementation(console.log),
+    // tslint:disable-next-line: no-console
+    debug: jest.fn().mockImplementation(console.log)
+  } as unknown as Logger;
+
+  it("should return a right response",async () => {
+      var handler = InfoControllers.getInfo(config, logger);
+
+      var req2 = {} as any;
```
Comment (giovanniberti) [inline src/controllers/__tests__/InfoControllers.test.ts:24]: Const?

File: src/controllers/__tests__/InfoControllers.test.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+      const responseErrorValidation2 = await handler(req2);
+
+      expect(responseErrorValidation2.kind).toBe("IResponseSuccessJson");
```
Comment (giovanniberti) [inline src/controllers/__tests__/InfoControllers.test.ts:28]: Naming

```suggestion
      var request = {} as any;

      const response = await handler(request);

      expect(response.kind).toBe("IResponseSuccessJson");
```

File: src/controllers/__tests__/EmailsControllers.test.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    const response = await handler(getReq("success","CLIENT_ECOMMERCE"));
+    expect(mockSendMail).toThrowError();
+    expect(response.kind).toBe("IResponseSuccessJson");
+  });
+
+});
+
+describe("test template", () => {
+
+  var browser: Browser;
```
[the remark this answers is not in the document]
Comment (ciuffagianluca) [reply src/controllers/__tests__/EmailsControllers.test.ts:165]: I used it in more than one test and initialize it in beforeEach. Unless some you think i have to put it in each test, I think it should remain var

**Comment by ciuffagianluca** [comment]
> Good job 🚀 Left a few minor comments 😃

Hi Giovanni, all hints have been addressed except for one where I left a comment. 

Thank you for your review.

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**